src/main/java/org/jenkinsci/plugins/registry/notification/webhook/JSONWebHook.java

```diff
@@ -142,7 +143,7 @@ public void doIndex(StaplerRequest request, StaplerResponse response) throws IOE
         response.sendRedirect(request.getContextPath() + "/");
     }
 
-    protected abstract WebHookPayload createPushNotification(JSONObject data);
+    protected abstract WebHookPayload createPushNotification(JSONObject data, StaplerRequest request);
```

Comment:
normally changing a method signature like this is frowned upon. The old method should still be there and call the new method with sensible defaults or null.
In this case it is probably not needed, but easy enough to do, so up to you.